app/assets/javascripts/components/Glossary.js

```diff
@@ -0,0 +1,85 @@
+define(['jquery', 'DoughBaseComponent'], function($, DoughBaseComponent) {
+
+  'use strict';
+
+  var Glossary;
+
+  Glossary = function($el, config) {
+    Glossary.baseConstructor.call(this, $el, config);
+
+    var article = $('[data-dough-glossary-content]'),
+        articleContent = $(article).html(),
+        updatedContent;
+
+    $.getJSON('/glossary.json', function(data){
+      data.glossary.map(function (item) {
+
+        // If the article content contains any of the keywords in the JSON
+        if(articleContent.indexOf(item.term) >= 1){
+
+          console.log('The term ' + item.term + ' has been found in this article')
+
+          $(item).each(function(){
+            var updatedContent = articleContent.replace(RegExp("(?!<.*?)\\b(" + item.term + ")\\b(?![^<>]*?(<\/a>|<\/h2>|<\/h3>|>))","gi"),
+                                                               "<button class='unstyled-button glossary__term' aria-label='" + item.term + " (see description)'>" + item.term + "<sub>?</sub></a></button><span class='glossary__description'><span class='glossary__description-title'>" + item.term + "</a></span>" + item.description + "<button class='glossary__description-close'><svg xmlns='http://www.w3.org/2000/svg' \
```

Comment:
can a function be used here to enhance readability or even use a var for the regex?